lib/tool_shed/util/shed_util_common.py

```diff
@@ -98,20 +98,28 @@ def can_eliminate_repository_dependency(metadata_dict, tool_shed_url, name, owne
     return True
 
 
-def can_eliminate_tool_dependency(metadata_dict, name, type, version):
+def can_eliminate_tool_dependency(metadata_dict, name, dependency_type, version):
     """
     Determine if the relationship between a tool_dependency record
     associated with a tool_shed_repository record on the Galaxy side
     can be eliminated.
     """
     td_dict = metadata_dict.get('tool_dependencies', {})
     for td_key, td_val in td_dict.items():
-        n = td_val.get('name', None)
-        t = td_val.get('type', None)
-        v = td_val.get('version', None)
-        if n == name and t == type and v == version:
-            # The tool dependency is current, so keep it.
-            return False
+        if td_key == 'set_environment':
+            for td in td_val:
```

Comment:
@mvdbeek I recently tested that, and determined that the set_environment tool dependency would disappear if its value was updated, but the updated version would not be added.